From ecab12687091e0ab8da8865e7f639af73e4ba7da Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Wed, 12 Mar 2025 10:39:01 +0200
Subject: [PATCH] [SW-190031] boot_desc: rename boot_desc funcs

As a preparation for moving all funcs to emb_lib_cmn,
this patch renames the functions before removal

Change-Id: I0675c74407002082367ca23e471b5185e65700fe
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index ebc57dd..8dcdb4a 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -37,7 +37,7 @@
 	enum cpu_boot_err boot_err;
 	char *msg;
 
-	idx = boot_desc_get_img_desc(image_pending, HL_FW_IMG_PREBOOT, desc);
+	idx = _boot_desc_get_img_desc(image_pending, HL_FW_IMG_PREBOOT, desc);
 	if (idx == HL_FW_IMG_NOT_FOUND)
 		return false;
 
@@ -82,7 +82,7 @@
 	enum fw_post_reset_cmd image_pending = regmap_readl(global_conf.upd_pending_sts_map);
 
 	/* Initialize load address with illegal value */
-	if (!boot_desc_init()) {
+	if (!_boot_desc_init()) {
 		preboot_puts_raw("Not a valid boot desc.");
 		return false;
 	}
diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index f226681..96817ce 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -467,7 +467,7 @@
 	return hw_mem_copy(0, dst, size, is_hbw, true);
 }
 
-u32 dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
+u32 _dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
 			  struct boot_desc *desc)
 {
 	/* TODO: use stable index when bootinfo implementation is final in arcboot */
@@ -648,14 +648,14 @@
 	u32 rst_src, err;
 	bool no_return;
 
-	bootdesc_cnt = boot_desc_init();
+	bootdesc_cnt = _boot_desc_init();
 	if (!bootdesc_cnt) {
 		preboot_puts_raw("No valid flash manifest");
 		return false;
 	}
 
 	for (i = 0; i < bootdesc_cnt; i++) {
-		err = boot_desc_get_img_desc_by_idx(i, &desc);
+		err = _boot_desc_get_img_desc_by_idx(i, &desc);
 		if (err) {
 			preboot_puts_hex("Failed to get boot descriptor no. ", i, false);
 			continue;
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index f545c47..0e8c427 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1386,7 +1386,7 @@
 #define BOOT_PTRS_NEXT_MAGIC	0x31313131
 #define BOOT_PTRS_MOD_MAGIC	0x30303030
 
-bool is_legacy_boot_descs(struct hl_fw_boot_ptrs *boot_ptrs)
+bool _is_legacy_boot_descs(struct hl_fw_boot_ptrs *boot_ptrs)
 {
 	int i;
 	uint32_t *magic = &boot_ptrs->next;
@@ -1400,7 +1400,7 @@
 	return true;
 }
 
-bool dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
+bool _dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
 {
 	if (img_desc->idx >= 2)
 		return false;
@@ -1408,7 +1408,7 @@
 	return true;
 }
 
-u32 dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
+u32 _dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
 			  struct boot_desc *desc)
 {
 	u32 idx = desc_ptr->idx;
diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 3b3664e..cfda72f 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -60,7 +60,7 @@
 	bool is_valid = true;
 	u32 idx;
 
-	idx = boot_desc_get_img_desc(FLOPS_UPD_NOOP, HL_FW_IMG_SRAM_MBIST_DATA, desc);
+	idx = _boot_desc_get_img_desc(FLOPS_UPD_NOOP, HL_FW_IMG_SRAM_MBIST_DATA, desc);
 	if (idx == HL_FW_IMG_NOT_FOUND)
 		return false;
 
@@ -87,7 +87,7 @@
 	bool ret;
 
 	/* Initialize load address with illegal value */
-	if (!boot_desc_init()) {
+	if (!_boot_desc_init()) {
 		preboot_puts_raw("Not a valid boot desc.");
 		return false;
 	}
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 32d0d64..07e92bd 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -777,7 +777,7 @@
 	bool is_psoc_mem;
 
 	img_type = data->img_type[data->img_idx];
-	idx = boot_desc_get_img_desc(FLOPS_UPD_NOOP, img_type, &desc);
+	idx = _boot_desc_get_img_desc(FLOPS_UPD_NOOP, img_type, &desc);
 	if (idx == HL_FW_IMG_NOT_FOUND) {
 		preboot_puts("image not found, err loading pif fw\n");
 		return -1;
@@ -905,7 +905,7 @@
 	regmap_writel(PIF_ARC_ARC_RST_PRESETDBGN_M, rst_reg);
 
 	if (hl_rst_src_get() == HL_COLD_RST) {
-		if (!boot_desc_init()) {
+		if (!_boot_desc_init()) {
 			preboot_puts("No boot desc, err loading pif fw\n");
 			return;
 		}
@@ -1286,12 +1286,12 @@
 	if (rst_src != HL_COLD_RST)
 		return;
 
-	if (!boot_desc_init()) {
+	if (!_boot_desc_init()) {
 		preboot_puts("No boot desc, err loading Phy FW\n");
 		return;
 	}
 
-	idx = boot_desc_get_img_desc(FLOPS_UPD_NOOP, HL_FW_IMG_PPBOOT, &desc);
+	idx = _boot_desc_get_img_desc(FLOPS_UPD_NOOP, HL_FW_IMG_PPBOOT, &desc);
 	if (idx == HL_FW_IMG_NOT_FOUND) {
 		preboot_puts("ppboot descriptor, err loading Phy FW\n");
 		return;
@@ -1682,7 +1682,7 @@
 		}
 }
 
-bool dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
+bool _dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
 {
 	if (img_desc->idx >= 2)
 		return false;
@@ -1690,7 +1690,7 @@
 	return true;
 }
 
-u32 dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
+u32 _dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
 			  struct boot_desc *desc)
 {
 	u32 idx = desc_ptr->idx;
diff --git a/include/boot_desc.h b/include/boot_desc.h
index 6ad8792..182e71e 100644
--- a/include/boot_desc.h
+++ b/include/boot_desc.h
@@ -34,14 +34,14 @@
 };
 
 /*
- * boot_desc_init() - read and initialize the boot image pointers structures
+ * _boot_desc_init() - read and initialize the boot image pointers structures
  *
  * Returns: the amount of images available, or 0 - no images available - legacy
  */
-int boot_desc_init(void);
+int _boot_desc_init(void);
 
 /*
- * boot_desc_get_img_desc() - get the struct boot_desc of the requested image
+ * _boot_desc_get_img_desc() - get the struct boot_desc of the requested image
  *
  * @upd_status	- FLOPS_UPD_PENDING - for modifiable image offset,
  *		  other values - for stable image offset
@@ -52,13 +52,13 @@
  * On success: the index of the image to load
  * On failure: (u32)-1 (no image by the requested type)
  */
-u32 boot_desc_get_img_desc(enum fw_post_reset_cmd upd_status,
+u32 _boot_desc_get_img_desc(enum fw_post_reset_cmd upd_status,
 			   enum hl_fw_img_type type, struct boot_desc *desc);
-u32 boot_desc_get_img_desc_by_idx(unsigned int idx, struct boot_desc *desc);
+u32 _boot_desc_get_img_desc_by_idx(unsigned int idx, struct boot_desc *desc);
 
-bool is_legacy_boot_descs(struct hl_fw_boot_ptrs *boot_ptrs);
-bool dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc);
-u32 dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
+bool _is_legacy_boot_descs(struct hl_fw_boot_ptrs *boot_ptrs);
+bool _dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc);
+u32 _dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
 			  struct boot_desc *desc);
 
 #endif /* _BOOT_DESC_H_ */
diff --git a/lib/common/boot_desc.c b/lib/common/boot_desc.c
index d5a7e18..04a511e 100644
--- a/lib/common/boot_desc.c
+++ b/lib/common/boot_desc.c
@@ -26,9 +26,9 @@
 static u8 desc_buf[BOOT_DESC_MAX_IMGS * IMG_DESC_SZ];
 static bool desc_initialized;
 
-__weak bool dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc) { return true; }
+__weak bool _dev_spec_img_desc_is_valid(struct hl_fw_img_desc *img_desc) { return true; }
 
-static bool boot_desc_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
+static bool _boot_desc_img_desc_is_valid(struct hl_fw_img_desc *img_desc)
 {
 	boot_desc_print_magic("FW IMG DESC", &img_desc->magic);
 
@@ -38,14 +38,14 @@
 	if (img_desc->desc_ver == 0 || img_desc->desc_ver == ~0)
 		return false;
 
-	return dev_spec_img_desc_is_valid(img_desc);
+	return _dev_spec_img_desc_is_valid(img_desc);
 }
 
 __weak int ppboot_boot_info_get(u32 type, void *data, u32 max_size, u32 *copy_size) { return 1; }
 
 #ifdef READ_BOOT_DESC_FROM_BOOTINFO
 
-int boot_desc_init(void)
+int _boot_desc_init(void)
 {
 	struct hl_fw_desc_info desc_info;
 	uint32_t copy_sz;
@@ -81,7 +81,7 @@
 
 	for (i = 0; i < boot_ptrs.count; i++) {
 		desc_ptr = (struct hl_fw_img_desc *)(desc_buf + i * boot_ptrs.img_desc_sz);
-		if (!boot_desc_img_desc_is_valid(desc_ptr)) {
+		if (!_boot_desc_img_desc_is_valid(desc_ptr)) {
 			preboot_puts_hex_raw("Bad boot descriptor: ", i, false);
 			return 0;
 		}
@@ -94,11 +94,11 @@
 
 #else
 
-__weak bool is_legacy_boot_descs(struct hl_fw_boot_ptrs *boot_ptrs) { return false; }
+__weak bool _is_legacy_boot_descs(struct hl_fw_boot_ptrs *boot_ptrs) { return false; }
 
 #define LEGACY_IMAGE_DESCRIPTOR_SIZE	220
 
-int boot_desc_init(void)
+int _boot_desc_init(void)
 {
 	unsigned int i;
 	u32 spi_off = HL_FW_BOOT_PTRS_BASE;
@@ -120,7 +120,7 @@
 		return 0;
 
 	/* legacy descriptors don't have size field. so constant value is used */
-	if (is_legacy_boot_descs(&boot_ptrs))
+	if (_is_legacy_boot_descs(&boot_ptrs))
 		boot_ptrs.img_desc_sz = LEGACY_IMAGE_DESCRIPTOR_SIZE;
 
 	if (boot_ptrs.count * boot_ptrs.img_desc_sz > sizeof(desc_buf)) {
@@ -140,7 +140,7 @@
 		spi_off = HL_FW_BOOT_PTRS_BASE + desc_offs[i];
 		buff_ptr = desc_buf + i * boot_ptrs.img_desc_sz;
 		spi_flash_copy(buff_ptr, spi_off, boot_ptrs.img_desc_sz, "IMG DESC");
-		if (!boot_desc_img_desc_is_valid((struct hl_fw_img_desc *)buff_ptr)) {
+		if (!_boot_desc_img_desc_is_valid((struct hl_fw_img_desc *)buff_ptr)) {
 			preboot_puts_hex_raw("Bad boot descriptor: ", i, false);
 			return 0;
 		}
@@ -153,7 +153,7 @@
 
 #endif /* BOOT_DESC_SRC_BOOT_INFO */
 
-__weak u32 dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
+__weak u32 _dev_spec_get_img_desc(struct hl_fw_img_desc *desc_ptr, enum fw_post_reset_cmd upd_status,
 				 struct boot_desc *desc)
 {
 	preboot_puts("error: dev spec get img desc not implemented\n");
@@ -161,7 +161,7 @@
 	return HL_FW_IMG_NOT_FOUND;
 }
 
-u32 boot_desc_get_img_desc(enum fw_post_reset_cmd upd_status,
+u32 _boot_desc_get_img_desc(enum fw_post_reset_cmd upd_status,
 			   enum hl_fw_img_type type, struct boot_desc *desc)
 {
 	int i, ret;
@@ -173,7 +173,7 @@
 		if (desc_ptr->magic.type != type)
 			continue;
 
-		ret = dev_spec_get_img_desc(desc_ptr, upd_status, desc);
+		ret = _dev_spec_get_img_desc(desc_ptr, upd_status, desc);
 
 		return ret;
 	}
@@ -182,7 +182,7 @@
 	return HL_FW_IMG_NOT_FOUND;
 }
 
-u32 boot_desc_get_img_desc_by_idx(unsigned int idx, struct boot_desc *desc)
+u32 _boot_desc_get_img_desc_by_idx(unsigned int idx, struct boot_desc *desc)
 {
 	struct hl_fw_img_desc *desc_ptr;
 
@@ -191,5 +191,5 @@
 
 	desc_ptr = (struct hl_fw_img_desc *)(desc_buf + idx * boot_ptrs.img_desc_sz);
 
-	return dev_spec_get_img_desc(desc_ptr, FLOPS_UPD_NOOP, desc);
+	return _dev_spec_get_img_desc(desc_ptr, FLOPS_UPD_NOOP, desc);
 }
diff --git a/lib/preboot/i2cslave_indirect.c b/lib/preboot/i2cslave_indirect.c
index b3fe54a..18d10dd 100644
--- a/lib/preboot/i2cslave_indirect.c
+++ b/lib/preboot/i2cslave_indirect.c
@@ -487,7 +487,7 @@
 	spi->ppboot_primary = !hl_ppboot_is_valid(true);
 	spi->ppboot_secondary = !hl_ppboot_is_valid(false);
 
-	if (!boot_desc_init()) {
+	if (!_boot_desc_init()) {
 		preboot_puts_raw("Not a valid boot desc.");
 		return;
 	}
@@ -495,7 +495,7 @@
 	preboot_get_mem_region(MEM_I2C, &mdesc);
 
 	for (i = 0; i < 2; i++) {
-		idx = boot_desc_get_img_desc(role[i], HL_FW_IMG_PREBOOT, &desc);
+		idx = _boot_desc_get_img_desc(role[i], HL_FW_IMG_PREBOOT, &desc);
 		if (idx == HL_FW_IMG_NOT_FOUND)
 			continue;
 
